core/lib/types/src/commitment/mod.rs

```diff
@@ -606,7 +593,9 @@ impl L1BatchCommitment {
         let mut result = vec![];
         let pass_through_data_hash = self.pass_through_data.hash();
         result.extend_from_slice(pass_through_data_hash.as_bytes());
-        let metadata_hash = self.meta_parameters.hash();
+        let metadata_hash = self
+            .meta_parameters
+            .hash(&self.auxiliary_output.common().protocol_version);
```

Comment:
`meta_parameters` and `auxiliary_output` should be independent. I suggest adding `protocol_version` field to `L1BatchMetaParameters`